modules/apps/portal-vulcan/portal-vulcan-impl/src/main/java/com/liferay/portal/vulcan/internal/exportimport/data/handler/VulcanBatchEnginePortletDataHandlerRegistry.java

```diff
@@ -0,0 +1,176 @@
+				bundleContext));
+
+		_serviceRegistration = bundleContext.registerService(
+			FeatureFlagListener.class,
+			(companyId, featureFlagKey, enabled) -> {
+				if (enabled) {
+					_serviceTracker.open();
+				}
+				else {
+					_serviceTracker.close();
+				}
+			},
+			MapUtil.singletonDictionary("featureFlagKey", "LPD-35914"));
+	}
+
+	@Deactivate
+	protected void deactivate() {
+		_serviceRegistration.unregister();
+		_serviceTracker.close();
+
+		for (ServiceRegistration<PortletDataHandler> serviceRegistration :
+				_serviceRegistrations.values()) {
+
+			try {
+				serviceRegistration.unregister();
+			}
+			catch (IllegalStateException illegalStateException) {
+				_log.error(illegalStateException);
+			}
+		}
+
+		_serviceRegistrations.clear();
+	}
+
+	private static final Log _log = LogFactoryUtil.getLog(
+		VulcanBatchEnginePortletDataHandlerRegistry.class);
+
+	@Reference
+	private BatchEngineExportTaskExecutor _batchEngineExportTaskExecutor;
+
+	@Reference
+	private BatchEngineExportTaskService _batchEngineExportTaskService;
+
+	@Reference
+	private BatchEngineImportTaskExecutor _batchEngineImportTaskExecutor;
+
+	@Reference
+	private BatchEngineImportTaskService _batchEngineImportTaskService;
+
+	private ServiceRegistration<FeatureFlagListener> _serviceRegistration;
+	private final Map
+		<ServiceReference<VulcanBatchEngineTaskItemDelegate>,
+		 ServiceRegistration<PortletDataHandler>> _serviceRegistrations =
+			new ConcurrentHashMap<>();
+	private ServiceTracker
+		<VulcanBatchEngineTaskItemDelegate, PortletDataHandler> _serviceTracker;
+
+	private class VulcanBatchEngineTaskItemDelegateServiceTrackerCustomizer
+		implements ServiceTrackerCustomizer
+			<VulcanBatchEngineTaskItemDelegate, PortletDataHandler> {
+
+		public VulcanBatchEngineTaskItemDelegateServiceTrackerCustomizer(
+			BundleContext bundleContext) {
+
+			_bundleContext = bundleContext;
+		}
+
+		@Override
+		public PortletDataHandler addingService(
+			ServiceReference<VulcanBatchEngineTaskItemDelegate>
+				serviceReference) {
+
+			String portletId = (String)serviceReference.getProperty(
+				"batch.engine.task.item.delegate.portlet.id");
+
+			if (Validator.isNull(portletId)) {
+				return null;
+			}
+
+			VulcanBatchEnginePortletDataHandler
+				vulcanBatchEnginePortletDataHandler =
+					new VulcanBatchEnginePortletDataHandler(
+						_batchEngineExportTaskExecutor,
+						_batchEngineExportTaskService,
+						_batchEngineImportTaskExecutor,
+						_batchEngineImportTaskService,
+						(String)serviceReference.getProperty(
+							"batch.engine.task.item.delegate.class.name"),
+						(String)serviceReference.getProperty(
+							"batch.engine.task.item.delegate.name"));
+
+			_serviceRegistrations.put(
+				serviceReference,
+				_bundleContext.registerService(
+					PortletDataHandler.class,
+					vulcanBatchEnginePortletDataHandler,
+					HashMapDictionaryBuilder.<String, Object>put(
+						"javax.portlet.name", portletId
```

Comment:
Well, I tested manually and it seems we are managing it properly based on the company group, so I think we should forget this comment

Reply:
Staging already filters the portlets by company id, and the ObjectEntriesPortlet is already scoped. Since the PortletDataHandlers are registered per portlet it should be fine.